test/services/caServiceTest.js

```diff
@@ -0,0 +1,74 @@
+const createCaService = require('../../server/services/caService')
+
+describe('caService', () => {
+  let roService
+  let caService
+  const lduActiveClient = { isLduPresent: () => {} }
+
+  describe('Responsible Officer is allocated and Ldu determined', () => {
+    const responsibleOfficerOrError = {
+      isAllocated: true,
+      lduCode: 'ldu-123',
+    }
+
+    beforeEach(() => {
+      roService = {
+        findResponsibleOfficer: sinon.stub().resolves(responsibleOfficerOrError),
+      }
+    })
+
+    describe('getReasonForNotContinuing ', () => {
+      it('should return null because Ro is COM and ldu is active', async () => {
+        lduActiveClient.isLduPresent = sinon.stub().resolves(1)
+        caService = createCaService(roService, lduActiveClient)
+
+        const result = await caService.getReasonForNotContinuing('bookingId-1', 'token-1')
+        expect(result).to.eql(null)
+      })
+
+      it('should return LDU_INACTIVE because ldu is not active', async () => {
+        lduActiveClient.isLduPresent = sinon.stub().resolves(0)
+        caService = createCaService(roService, lduActiveClient)
+
+        const result = await caService.getReasonForNotContinuing('bookingId-1', 'token-1')
+        expect(result).to.eql('LDU_INACTIVE')
+      })
+
+      it('should return COM_NOT_ALLOCATED because COM has not been assigned', async () => {
+        lduActiveClient.isLduPresent = sinon.stub().resolves(1)
+        caService = createCaService(roService, lduActiveClient)
+
+        responsibleOfficerOrError.isAllocated = false
+        const result = await caService.getReasonForNotContinuing('bookingId-1', 'token-1')
+        expect(result).to.eql('COM_NOT_ALLOCATED')
+        responsibleOfficerOrError.isAllocated = true
+      })
+    })
+  })
+
+  describe('Responsible officer not assigned or Ldu not determined', () => {
+    const responsibleOfficerOrError = {
```

Comment:
and maybe just always call this `Error`, as it always is, in this set of tests

Reply:
done